cfme/cloud/availability_zone.py

```diff
@@ -1,74 +1,173 @@
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table("//div[@id='list_grid']//table")
+    search = View.nested(Search)
+
+
+class AvailabilityZoneDetailsEntities(View):
+    """View containing the widgets for the main content pane on the details page"""
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    relationships = SummaryTable(title='Relationships')
+    smart_management = SummaryTable(title='Smart Management')
+
+
+class AvailabilityZoneDetailsAccordion(View):
+    """View containing the accordion widgets for the left side pane on details view"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
 
-# Page specific locators
-listview_pagetable = PagedTable(table_locator="//div[@id='list_grid']//table")
-listview_checktable = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
-pol_btn = partial(tb.select, 'Policy')
-mon_btn = partial(tb.select, 'Monitoring')
+class AvailabilityZoneView(BaseLoggedInPage):
+    """Bare bones base view for page header matching"""
+    @property
+    def in_availability_zones(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Compute', 'Clouds', 'Availability Zones'] and
+            match_location(controller='availability_zone', title='Availability Zones'))
+
+
+class AvailabilityZoneAllView(AvailabilityZoneView):
+    """Collect the view components into a single view"""
+    @property
+    def is_displayed(self):
+        return(
+            self.in_availability_zones and
+            self.entities.title.text == 'Availability Zones')
+
+    toolbar = View.nested(AvailabilityZoneToolBar)
+    entities = View.nested(AvailabilityZoneEntities)
+    paginator = View.nested(PaginationPane)
+
 
-match_page = partial(match_location, controller='availability_zone', title='Availability Zones')
+class AvailabilityZoneDetailsView(AvailabilityZoneView):
+    """Collect the view components into a single view"""
+    @property
+    def is_displayed(self):
+        expected_title = "{} (Summary)".format(self.context['object'].name)
+        expected_detail = self.context['object'].provider.name
+        return (
+            self.in_availability_zones and
+            self.entities.title.text == expected_title and
+            self.entities.relationships.get_text_of('Cloud Provider') == expected_detail)
+
+    toolbar = View.nested(AvailabilityZoneDetailsToolBar)
+    sidebar = View.nested(AvailabilityZoneDetailsAccordion)
+    entities = View.nested(AvailabilityZoneDetailsEntities)
+
+
+class AvailabilityZoneEditTagsView(AvailabilityZoneView):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h3')
+    # TODO Add tag table support when rowspan is supported in SummaryTable
```

Comment:
Probably raise an issue in widgetastic.core and we'll have to brainstorm how to handle it. This will go in the base table and then SummaryTable will be just slightly tweaked.